src/metatensor/models/utils/data/dataset.py

```diff
@@ -28,15 +28,15 @@ def compiled_join(a, axis, remove_tensor_name):
     compiled_join = torch.jit.script(metatensor.torch.join)
 
 
-def get_all_species(dataset: _BaseDataset) -> List[int]:
+def get_all_species(datasets: List[_BaseDataset]) -> List[int]:
```

Comment:
Maybe we allow a list and a single `_BaseDataset` and internally convert? Or even better do it as a `*args` parameter.